language/move-lang/src/typing/globals.rs

```diff
@@ -223,110 +217,94 @@ fn builtin_function(
 
 fn check_acquire_listed<F>(
     context: &mut Context,
-    annotated_acquires: &BTreeSet<BaseType>,
-    loc: &Loc,
+    annotated_acquires: &BTreeSet<StructName>,
+    loc: Loc,
     msg: F,
-    global_type: &BaseType,
+    global_type: &StructName,
 ) where
     F: Fn() -> String,
 {
     if !annotated_acquires.contains(global_type) {
-        let ty_debug = global_type.value.subst_format(&HashMap::new());
-        context.error(vec![
-            (*loc, msg()),
-            (global_type.loc, format!("The call acquires '{}', but the 'acquires' list for the current function does not contain this type. It must be present in the calling context's acquires list", ty_debug))
-        ]);
+        let tmsg = format!(
+            "The call acquires '{}::{}', but the 'acquires' list for the current function does not contain this type. It must be present in the calling context's acquires list",
+            context.current_module.as_ref().unwrap(),
+            global_type
+        );
+        context.error(vec![(loc, msg()), (global_type.loc(), tmsg)]);
     }
 }
 
-pub fn check_global_access<F>(
+pub fn check_global_access<'a, F>(
     context: &mut Context,
     loc: &Loc,
     msg: F,
-    global_type: &BaseType,
-) -> bool
+    global_type: &'a BaseType,
+) -> Option<&'a StructName>
 where
     F: Fn() -> String,
 {
-    check_global_access_(context, loc, msg, global_type, true)
+    check_global_access_(context, loc, msg, global_type)
 }
 
-fn valid_acquires_annot(context: &mut Context, loc: &Loc, global_type: &BaseType) -> bool {
-    let msg = || panic!("ICE should not have recorded errors");
-    check_global_access_(context, loc, msg, global_type, false)
+fn valid_acquires_annot(_context: &mut Context, _loc: Loc, _global_type: &StructName) -> bool {
```

Comment:
Kill?

Reply:
I wanted to leave the flow in place as there might be some changes to these rules in the future